File: app/src/main/java/com/example/test_store/Logowanie/LoginContract.java
package com.example.test_store.Logowanie;

import android.view.View;

public interface LoginContract {
    interface View{
        void onLoginClick(android.view.View v);
        void onRegisterClick(android.view.View v);
        void showToast(String message);
    }

    interface Presenter{
        void checkIfLoggedIn();
        boolean validateEmail(String email);
        boolean validatePassword(String password);
        void onLoginClicked();
        void openRegisterActivity();
    }
}

File: app/src/main/java/com/example/test_store/Logowanie/LoginContract.kt
package com.example.test_store.Logowanie

import android.view.View

interface LoginContract {
    interface View {
        fun onLoginClick(v: android.view.View?)
        fun onRegisterClick(v: android.view.View?)
        fun showToast(message: String?)
    }

    interface Presenter {
        fun checkIfLoggedIn()
        fun validateEmail(email: String?): Boolean
        fun validatePassword(password: String?): Boolean
        fun onLoginClicked()
        fun openRegisterActivity()
    }
}
